src/Microsoft.VisualStudio.LanguageServices.Razor/Editor/DefaultVisualStudioDocumentTracker.cs

```diff
@@ -129,58 +138,17 @@ public void Subscribe()
         public void Unsubscribe()
         {
             _projectManager.Changed -= ProjectManager_Changed;
+
+            // Detached from project.
+            _isSupportedProject = false;
+            _project = null;
+            OnContextChanged(project: null);
```

Comment:
Notify listeners on unsubscribe so the parser can shut down its internal background parser.

Reply:
??? We need to discuss this

Reply:
Talked offline, this is fine.